### ISSUE TITLE
about output loss definition

### ISSUE THREAD

**Issue by hinanmu** [dhruvramani/C2AE-Multilabel-Classification#8]
Hi,I have a question in your code about loss definition. I notice that the prediction of the output loss in the C2AE paper is defined as F_d(F_e(x_i)). But it is defined as F_d(F_x(x_i)) in your code.

Is it a mistake in the paper? I don't know why F_e function can accept x_i, they have a different dimension.
Thank you!

**Comment by greeness** [comment]
The prediction is define here:
https://github.com/dhruvramani/C2AE-Multilabel-Classification/blob/master/model/src/network.py#L72,

where prediction = F_d(F_x(X)). Actually I think this is the one define in the paper (see the last sentence in section "Canonical-Correlated Autoencoder (C2AE)", 

To be more precise, a test input xˆ will be first transformed into the derived latent space by Fx, followed by the decoding mapping of Fd for predicting its output label yˆ (i.e., yˆ = Fd(Fx(xˆ)).

The output loss is defined as a weighted sum between two components:
object or loss = loss(Fx, Fe) + alpha * loss (Fe, Fd)

1. the loss at the embedding space - loss (Fx, Fe)
2. the output of C2AE - loss(Fe, Fd). 

I am not sure where you see that the output loss is defined as F_d(F_e(x_i)).

**Comment by greeness** [comment]
I think there is a mistake in the paper. 

In the training time, to get the loss between Fe and Fd, we need to calculate Fd(Fe(y_i)) (Not x_i, because as you pointed out, Fe cannot take x_i as input but y_i). 

Probably because of this mistake, dhruvramani@ used a different way to derive the loss. It is probably working too but not as good as the one proposed by the original paper.

In the prediction time, to predict labels, we would do

Fd(Fx(x_i)) this is consistent between the formula from the paper and the implementation here.

**Comment by greeness** [comment]
I found out that in the latest version of the paper hosted on arXiV . That mistake is already fixed. Now equation (3) cotains Fd(Fe(y_i)) instead of Fd(Fe(x_i)). 

See https://arxiv.org/pdf/1707.00418.pdf

**Comment by hinanmu** [comment]
Thanks, I got it.

**Comment by chihkuanyeh** [comment]
@greeness thanks for the clarification.
